### PR TITLE
Karel/add substrace

### PR DESCRIPTION
**Pull request by KaiserKarel**
Adds a custom linter which checks if pallets have the proper clippy attributes to avoid panics.

### CODE REVIEW

**Comment by cor** [comment]
The linter looks very nice, but why are there changes to `runtime/composable/src/lib.rs`?

File: Cargo.toml [hunk 1]
```diff
@@ -207,3 +207,8 @@ pallet-beefy-mmr = { git = "https://github.com/composableFi/substrate", branch =
 beefy-merkle-tree = { git = "https://github.com/composableFi/substrate", branch = "polkadot-v0.9.16" }
 pallet-nicks = { git = "https://github.com/composableFi/substrate", branch = "polkadot-v0.9.16" }
 pallet-staking-reward-fn = { git = "https://github.com/composableFi/substrate", branch = "polkadot-v0.9.16" }
+
+[workspace.metadata.dylint]
+libraries = [
+    { git = "https://github.com/kaiserkarel/substrace" },
```
Comment (dzmitry-lahoda) [inline Cargo.toml:213]: ok. so would prefer this to be part of composable repo for easy maintenance until tool will become stable amid several upgrade of substrate.

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by seunlanlege — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (KaiserKarel) [reply Cargo.toml:213]: Yeah I can fork the repo to composable, do dev on my branch then sync regularly for example, or just migrate it over entirely.

**Comment by KaiserKarel** [comment]
Just need to wait for sccache to be removed from the CI-pipeline for this to pass CI I think.

**Comment by KaiserKarel** [comment]
> Substrace looks very nice, but why are there changes to `runtime/composable/src/lib.rs`?

Due to how some of the substrate macros expand, `impl_runtime_apis` adds `panic!` without putting the expanded code in module with a tag indicating it came from a macro (at least seems like it). This caused the `clippy::panic` lint to fail as `panics` were present in the expanded codebase. 

Originally we just disabled that warn, as the module generated did not export `RUNTIME_API_VERSIONS`, so you could not put the macro specifically in it's own module and allow `clippy::panic` there.

Now with some slight code reorgs, it is actually possible, meaning that we only allow `panic` for the macro expanded code, and lint for it in other sections. Before this someone could do:


```
ExistentialDeposit = || { panic!("think of a deposit") } 
```

In the runtime and actually have the code pass checks.

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -232,10 +232,15 @@ jobs:
...
+        run: |
+          cargo install cargo-dylint dylint-link
+          RUSTC_WRAPPER="" SKIP_WASM_BUILD=1 cargo dylint substrace --workspace    
+      
       - name: Cargo clippy
         run: |
-          SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --all-targets --tests -- -D warnings
-
+          SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --workspace --all-targets --tests -- -D warnings
+
```
Comment (dzmitry-lahoda) [inline .github/workflows/check.yml:243]: spaces, may be make style

**Pull request closed without merging**